### ISSUE TITLE
[FEATURE] Server side and client side cron jobs

### ISSUE THREAD

**Issue by akbansa** [appsmithorg/appsmith#738]
## Summary

So let's say I want to send a report based on some queries + client-side logic, then I need to run a cron job that will run client-side code and then send a report.

## Motivation

Automate a few reports and push data to different services like slack/intercom (in a nutshell call a webhook)

**Comment by kevev** [comment]
A way to schedule data source refresh would be a great addition as it would enable live dashboards.

**Comment by samg666** [comment]
Being able to run appsmith apps in background like cron job is really helpful. In my case, I will need to regularly check if any new orders via the API, if yes then get the order details and sync to my another MySQL DB. The current workaround is use Interval to regularly run the query, but I will prefer it can be run in background.

**Comment by ame-appsmith** [comment]
[This user](https://app.intercom.com/a/apps/y10e7138/inbox/inbox/conversation/164629100162180#part_id=comment-164629100162180-16719000134) also requested this feature.

**Comment by alienfacepalm** [comment]
Agreed. Airplane.dev does this out of the box. We need this feature so we can schedule jobs to be run for when customers meet certain thresholds we want the customer success team to interject, but without these jobs, they are operating in the dark.

**Comment by ame-appsmith** [comment]
[A user](https://app.frontapp.com/open/msg_1oc7m170?key=enofnivP2Lj8lYrJ5eWUZvVeqqqWitPg) is looking for a built-in scheduler. They want to schedule their API calls to happen once a day without interaction.

**Comment by bayeslearner** [comment]
only reason to choose budibase, ui-bakery over appsmith.

**Comment by M-Markgraf** [comment]
Cron Jobs or scheduled jobs inside appsmith running background would ahuge advantage for the use of management systems

**Comment by riteshkew** [comment]
Would be covered as part of Workflows.
